symbolic-base/test/Tests/Arithmetization/Optimization.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE TypeOperators       #-}
+
+module Tests.Arithmetization.Optimization (specOptimization) where
+
+import           Data.Binary                                (Binary)
+import           GHC.Generics                               (Par1 (..), U1 (..), type (:*:))
+import           Prelude                                    (IO, Show, return, ($))
+import           Test.Hspec
+import           Test.QuickCheck.Property                   ((.&.), (===))
+
+import           ZkFold.Base.Algebra.Basic.Class
+import           ZkFold.Base.Algebra.Basic.Number           (Natural)
+import           ZkFold.Symbolic.Class
+import           ZkFold.Symbolic.Compiler                   (compile)
+import           ZkFold.Symbolic.Compiler.ArithmeticCircuit
+import           ZkFold.Symbolic.Data.Bool                  (Bool (..))
+import           ZkFold.Symbolic.MonadCircuit
+
+
+testFunc :: forall a . (Arithmetic a, Binary a) => ArithmeticCircuit a U1 Par1 Par1
+testFunc = fromCircuitF idCircuit $ \(Par1 i0) -> do
+    i1 <- newRanged (fromConstant (1 :: Natural)) (at i0)
+    i2 <- newAssigned (($ i1) + one)
+    i3 <- newAssigned (($ i2) + one)
+    i4 <- newAssigned (($ i3) + one)
+    i5 <- newAssigned (($ i4) + ($ i0))
+    constraint (($ i4) - fromConstant (4 :: Natural))
+    return (Par1 i5)
+
+testBool :: forall a . (Arithmetic a, Binary a) => ArithmeticCircuit a (U1 :*: U1) (Par1 :*: U1) Par1
+testBool = ac
+    where
+        identBool :: Bool c -> Bool c
+        identBool x = x
+
+        Bool ac = compile @a identBool
```

Comment:
You can just `testBool = compile @a identBool`, no need to wrap it in `Bool` and unwrap back